config.go

```diff
@@ -26,6 +27,24 @@ type defaultsHandler interface {
 	SetDefaults(c *Config) error
 }
 
+// Define backoffStrategy
+type BackoffStrategy interface {
+   Calculate(attempt int) time.Duration
+}
+
+type ExponentialStrategy struct {
```

Comment:
I think we should also provide `FullJitterStrategy` as an exported part of the package?